client/src/service.rs

```diff
@@ -65,22 +65,22 @@ where
 {
     type Command = p2p::Command;
 
-    fn command(&mut self, cmd: Self::Command) {
+    fn command_received(&mut self, cmd: Self::Command) {
         // TODO: Commands shouldn't be handled by the inner state machine.
         self.machine.command(cmd)
     }
 }
 
-impl<T, F, P, C> StateMachine for Service<T, F, P, C>
+impl<T, F, P, C> PeerProtocol for Service<T, F, P, C>
 where
     T: BlockTree,
     F: filter::Filters,
     P: peer::Store,
     C: AdjustedClock<net::SocketAddr>,
 {
-    type Message = [u8];
-    type Event = p2p::Event;
-    type DisconnectReason = p2p::DisconnectReason;
+    type PeerMessage = [u8];
+    type Notification = p2p::Event;
+    type DisconnectDemand = p2p::DisconnectReason;
```

Comment:
in favor to keep `DisconnectReason`

Reply:
We already have different type with the same name and two types named exactly add a lot of ambiguity